plugins/synaptics-vmm9/README.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Plugin: Synaptics VMM9xxxx
+---
+
+## Introduction
+
+This plugin updates the VMM9xxxx MST devices from Synaptics.
+
+## Firmware Format
+
+The daemon will decompress the cabinet archive and extract a firmware blob in
+a packed binary file format.
+
+This plugin supports the following protocol ID:
+
+* `com.synaptics.mst`
+
+## GUID Generation
+
+These devices use the standard USB DeviceInstanceId values, e.g.
+
+* `USB\VID_06CB&PID_9430`
+
+These devices also use custom GUID values using the board ID value which is different for each
+customer and hardware design, e.g.
+
+* `USB\VID_06CB&PID_9430&BID_5678`
+
+## Update Behavior
+
+The device storage banks are erased, and each block from the firmware is written.
+The plugin then asks the device to calculate the EEPROM CRC and compares it with the CRC of the
+provided image. If written correctly, the firmware is activated.
+
+## Vendor ID Security
+
+The vendor ID is set from the USB vendor, in this instance set to `USB:0x06CB`
+
+## Quirk Use
+
+This plugin uses the following plugin-specific quirks:
+
+### Flags=manual-restart-required
```

Comment:
What's your goal? Let some devices update manually and others automatically?

If so I think you should add the flag for usable while updating when allowing the delayed activation and set up the delayed activation callback.

Reply:
> Let some devices update manually and others automatically?

Exactly that; some devices like the EVB we can just reboot the MST chip, and on others (where the MST chip doesn't control it's own power sequencing) we'll need to do a USB unplug/replug.

> when allowing the delayed activation

I don't know how well that's going to work when integrating this into other plugins like dell-dock -- it's certainly a lot simpler this way. I guess the other thing we could do is have a `Flags=manual-restart-required` *and* a `Flags=can-auto-restart` -- so that we can control the hub power sequencing in another plugin, perhaps in `composite_cleanup()` -- I'm hesitant to do that until we know the design of the "other" plugin.